miden-tx/src/verifier/mod.rs

```diff
@@ -0,0 +1,92 @@
+use super::{Digest, Hasher, TransactionComplier};
+use crypto::{WORD_SIZE, ZERO};
+use miden_core::{stack::STACK_TOP_SIZE, Felt, StackInputs, StackOutputs, Word};
+use miden_objects::{
+    notes::NoteEnvelope,
+    transaction::{ConsumedNoteInfo, ProvenTransaction},
+};
+use miden_verifier::{verify, VerificationError};
+
+/// The [TransactionVerifier] is used to verify a [ProvenTransaction].
+///
+/// The [TransactionVerifier] contains a [Kernel] object which we use to verify a given
+/// transaction against.
+pub struct TransactionVerifier {
+    compiler: TransactionComplier,
+}
```

Comment:
I wonder if we should add a field to mandate proof security level (e.g., `proof_security_level: u32`). This way, instead of returning it from the `verify()` method, we would reject the proofs which don't meet the required security level.

We'll probably make it a bit more sophisticated in the future, once [this issue](https://github.com/facebook/winterfell/issues/160#issuecomment-1563632829) is addressed in Winterfell.